pkg/controller/machine/controller.go

```diff
@@ -84,9 +97,8 @@ func (c *MachineControllerImpl) Reconcile(machine *clusterv1.Machine) error {
 			glog.Infof("reconciling machine object %v causes a no-op as there is no finalizer.", name)
 			return nil
 		}
-		// Master should not be deleted as part of reconcilation.
-		if cfg.ControllerConfig.InCluster && util.IsMaster(machine) {
-			glog.Infof("skipping reconciling master machine object %v", name)
+		if !c.isDeleteAllowed(machine) {
+			glog.Infof("skipping reconciling of machine object %v", name)
```

Comment:
Not your code but you should fix it while here: capitalize the first line of log messages (unlike error strings).

Reply:
Fixed this instance, will follow up in another PR for the other instances.